app/controllers/agent/predicates/ClientConfirmedController.scala

```diff
@@ -17,22 +17,26 @@
 package controllers.agent.predicates
 
 import auth.{MtdItUser, MtdItUserOptionNino, MtdItUserWithNino}
-import config.{AgentItvcErrorHandler, ItvcErrorHandler}
+import config.featureswitch.FeatureSwitching
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
 import controllers.agent.sessionUtils.SessionKeys
 import controllers.predicates.AuthPredicate.AuthPredicate
 import controllers.predicates.IncomeTaxAgentUser
 import controllers.predicates.agent.AgentAuthenticationPredicate
 import models.incomeSourceDetails.IncomeSourceDetailsModel
 import play.api.Logger
 import play.api.mvc.{AnyContent, MessagesControllerComponents, Request, Result}
-import services.IncomeSourceDetailsService
+import services.{IncomeSourceDetailsService, SessionDataService}
 import uk.gov.hmrc.auth.core.AffinityGroup.Agent
 import uk.gov.hmrc.auth.core.retrieve.Name
 import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException}
 
 import scala.concurrent.Future
 
-trait ClientConfirmedController extends BaseAgentController {
+trait ClientConfirmedController extends BaseAgentController with FeatureSwitching {
+
+  val appConfig: FrontendAppConfig
+  val sessionDataService: SessionDataService
```

Comment:
* no need to expose this on the level of trait, should be fine to add this as implicit for the methods consuming this service, this would reduce number of controllers affected by this change;